paimon-format/src/main/java/org/apache/paimon/format/orc/writer/OrcBulkWriter.java

```diff
@@ -67,4 +80,35 @@ public void finish() throws IOException {
         flush();
         writer.close();
     }
+
+    @Override
+    public long length() throws IOException {
+        long estimateMemory = treeWriter.estimateMemory();
+        long fileLength = underlyingStream.getPos();
+
+        // This value is estimated, not actual.
+        return (long) Math.ceil(fileLength + estimateMemory * 0.2);
```

Comment:
We don't need to read `stripes` now?

Reply:
For `stripes` that have been flushed, the size can be obtained through `underlyingStream.getPos()`, and for the stripe that is not fully formed, the size can be obtained through `estimateMemory()`.

Reply:
I see